hoprd/rest-api/src/session.rs

```diff
@@ -81,70 +128,289 @@ pub(crate) struct SessionClientResponse {
+                        ApiErrorStatus::UnknownFailure(format!("failed to start UDP listener on {bind_host}: {e}")),
+                    )
+                }
+            })?;
+
+            info!("UDP session listener bound to {bound_host}");
+
+            state.open_listeners.write().await.insert(
+                ListenerId(protocol, bound_host),
+                (
+                    target.clone(),
+                    hopr_async_runtime::prelude::spawn(bind_session_to_stream(session, udp_socket)),
+                ),
+            );
+            bound_host
+        }
+    };
+
+    Ok::<_, (StatusCode, ApiErrorStatus)>(
+        (
+            StatusCode::OK,
+            Json(SessionClientResponse {
+                protocol,
+                target,
+                ip: bound_host.ip().to_string(),
+                port: bound_host.port(),
+            }),
+        )
+            .into_response(),
+    )
+}
+
+/// Lists existing Session listeners for the given IP protocol.
+#[utoipa::path(
+    get,
+    path = const_format::formatcp!("{BASE_PATH}/session/{{protocol}}"),
+    params(
+            ("protocol" = String, Path, description = "IP transport protocol")
+    ),
+    responses(
+            (status = 200, description = "Opened session listeners for the given IP protocol.", body = Vec<SessionClientResponse>),
+            (status = 400, description = "Invalid IP protocol.", body = ApiError),
+            (status = 401, description = "Invalid authorization token.", body = ApiError),
+            (status = 422, description = "Unknown failure", body = ApiError)
+    ),
+    security(
+            ("api_token" = []),
+            ("bearer_token" = [])
+    ),
+    tag = "Session",
+)]
+pub(crate) async fn list_clients(
+    State(state): State<Arc<InternalState>>,
+    Path(protocol): Path<IpProtocol>,
+) -> Result<impl IntoResponse, impl IntoResponse> {
+    let response = state
+        .open_listeners
+        .read()
+        .await
+        .iter()
+        .filter(|(id, _)| id.0 == protocol)
+        .map(|(id, (target, _))| SessionClientResponse {
+            protocol,
+            target: target.clone(),
+            ip: id.1.ip().to_string(),
+            port: id.1.port(),
+        })
+        .collect::<Vec<_>>();
+
+    Ok::<_, (StatusCode, ApiErrorStatus)>((StatusCode::OK, Json(response)).into_response())
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[schema(example = json!({
+        "listeningIp": "127.0.0.1",
+        "port": 5542
+    }))]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionCloseClientRequest {
+    pub listening_ip: String,
+    pub port: u16,
+}
+
+/// Closes an existing Session listener.
+/// The listener must've been previously created and bound for the given IP protocol.
+/// Once a listener is closed, no more socket connections can be made to it.
+#[utoipa::path(
+    delete,
+    path = const_format::formatcp!("{BASE_PATH}/session/{{protocol}}"),
```

Comment:
In case the previous suggestion is applied, this is also needed.

```suggestion
    path = const_format::formatcp!("{BASE_PATH}/session/{{protocol}}/{{port}}"),
```

Reply:
See above.